conan/tools/cmake/cmakedeps/templates/target_data.py

```diff
@@ -215,8 +218,8 @@ def join_paths_single_var(values):
         # linker flags without magic: trying to mess with - and / =>
         # https://github.com/conan-io/conan/issues/8811
         # frameworks should be declared with cppinfo.frameworks not "-framework Foundation"
-        self.sharedlinkflags_list = join_flags(";", cpp_info.sharedlinkflags)
-        self.exelinkflags_list = join_flags(";", cpp_info.exelinkflags)
+        self.sharedlinkflags_list = join_flags(";", cpp_info.sharedlinkflags, as_string=True)
```

Comment:
This is a bit confusing. The ``self.xxxx_list`` variable was created to explicitly represent it contains a list of flags, not a unique string. Maybe we want to return to the ``sharedlinkflags`` variable? This was added because some CMake was complaining about receiving a string and it always needed a list, but better double check it.

Reply:
Yes, but the thing is that it's still a list of flags but passed as a string. I mean, internally, CMake is reading it as a list of flags -> `"oneflag;otherflag;-z flag"`.
I don't know what it was doing before but this is the recommended way to pass it, isn't it? For me, what could be confusing is the parameter `as_string`, any suggestion?

Reply:
I would remove the `as_string` and quote the final string in the assignation directly.